storage-proofs/src/hasher/types.rs

```diff
@@ -26,6 +24,7 @@ pub type PoseidonMDArity = U36;
 pub const MERKLE_TREE_ARITY: usize = 2;
 
 lazy_static! {
+    pub static ref POSEIDON_CONSTANTS_0: PoseidonConstants::<Bls12, U0> = PoseidonConstants::new();
```

Comment:
NOTE: although I understand why we do this, and it's clever that we can, we should add in a run-time check to ensure we *never* accidentally try to actually create a 0-ary hash. (My concern is that by making it possible to do it 'for real', any such accidental usage could survive as an invisible performance hit.)

Reply:
removing the constant, and panicing instead, safer